Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,408 @@
+            // Get a default trace configuration from a tier in the network.
+            guard let configuration = utilityNetwork
+                .definition?
+                .domainNetwork(named: "Pipeline")?
+                .tier(named: "Pipe Distribution System")?
+                .defaultTraceConfiguration
+            else { fatalError("Utility network does not have a default trace configuration.") }
+            if let category = category {
+                // Note: `UtilityNetworkAttributeComparison` or `UtilityCategoryComparison`
+                // with `UtilityCategoryComparisonOperator.doesNotExist` can also be used.
+                // These conditions can be joined with either `UtilityTraceOrCondition`
+                // or `UtilityTraceAndCondition`.
+                // See more in the README.
+                let comparison = UtilityCategoryComparison(category: category, operator: .exists)
+                // Create a trace filter.
+                let filter = UtilityTraceFilter()
+                filter.barriers = comparison
+                configuration.filter = filter
+            }
+            configuration.includesIsolatedFeatures = includesIsolatedFeatures
+            return configuration
+        }
+        
+        /// Adds a graphic at the tapped location for the filter barrier.
+        /// - Parameters:
+        ///   - feature: The geo element retrieved as a `Feature`.
+        ///   - location: The `Point` used to identify utility elements in the utility network.
+        func addFilterBarrier(for feature: ArcGISFeature, at location: Point) {
+            guard let geometry = feature.geometry,
+                  let element = utilityNetwork.makeElement(arcGISFeature: feature) else {
+                return
+            }
+            
+            switch element.networkSource.kind {
+            case .junction:
+                lastAddedElement = element
+                if let terminals = element.assetType.terminalConfiguration?.terminals {
+                    if terminals.count > 1 {
+                        terminalSelectorIsOpen.toggle()
+                        return
+                    } else {
+                        if let terminal = terminals.first {
+                            statusText = "Juntion element with terminal \(terminal.name) added to the filter barriers."
+                        }
+                    }
+                }
+            case .edge:
+                if let line = GeometryEngine.makeGeometry(from: geometry, z: nil) as? Polyline {
+                    element.fractionAlongEdge = GeometryEngine.polyline(
+                        line,
+                        fractionalLengthClosestTo: location,
+                        tolerance: -1
+                    )
+                    statusText = String(format: "Edge element at fractionAlongEdge %.3f added to the filter barriers.", element.fractionAlongEdge)
```

Comment:
Please don't reference Swift symbols in user-facing code. Also, as another example of why this should be done in the view, Apple's recommends using one of the newer formatting APIs instead of `String(format:...)` for user-facing formatted values.